### ISSUE TITLE
App model order for Admin Interface

### ISSUE THREAD

**Issue by saymoncoppi** [fabiocaccamo/django-admin-interface#139]
Hello Fabio!!!
Today I saw an improvement that fits with this old suggestion.
Beyond the foldable option could the django-admin-interface offer a way to reorder classes on tree menu. Take a look 
https://stackoverflow.com/questions/398163/ordering-admin-modeladmin-objects

![image](https://user-images.githubusercontent.com/22059715/140752908-888c6d82-d1d1-41f5-8fa5-f9ac28ba6ac1.png)

_Originally posted by @saymoncoppi in https://github.com/fabiocaccamo/django-admin-interface/issues/117#issuecomment-963169383_

**Comment by merwok** [comment]
Would be nice if this included hiding some app or specific models!

**Comment by fabiocaccamo** [comment]
It would be nice to manage all this from the admin itself.

**Comment by merwok** [comment]
Settings would be enough for me.  For me it’s about the app definition (for example, this project doesn’t use groups so don’t show them in admin), not appearance customization.  But I can see the other viewpoint!

**Comment by fabiocaccamo** [comment]
If you only need to hide Groups you can just unregister the ModelAdmin.

**Comment by merwok** [comment]
Ah of course 😄  The app and model would still be enabled, just hidden in admin.  Ignore my message!

**Comment by derzinn** [comment]
Just for the record: [django-modeladmin-reorder](https://github.com/mishbahr/django-modeladmin-reorder) works without problems in Django 4.1 and Python 3.10.

This package currently has no active maintainers, but an ownership change is [in progress](https://github.com/mishbahr/django-modeladmin-reorder/issues/45#issuecomment-1193968984).

**Comment by fabiocaccamo** [comment]
@derzinn cool, thank you for the suggestion!

**Comment by fabiocaccamo** [comment]
@derzinn I see the library you suggested has not been updated since many years and it has not tests, anyway it's a good reference for adding this feature directly into `django-admin-interface`.

**Comment by derzinn** [comment]
> @derzinn I see the library you suggested has not been updated since many years and it has not tests, anyway it's a good reference for adding this feature directly into `django-admin-interface`.

Yes, the codebase is a "little bit" outdated. :grin:  But runs fine. 

And it is very useful, for example:

```py
ADMIN_REORDER = (
    'sites',
    {
        'app': 'auth',
        'models': (
            'custom_auth.User',
            'auth.Group',
        )
    },
)
```

Including features like cross-linking and renaming models would be really cool! :+1:

**Comment by fabiocaccamo** [comment]
@derzinn indeed!

What do you think about making this configuration editable directly in the admin using json field? (it would require to add `django-json-widget` dependency).

**Comment by derzinn** [comment]
@fabiocaccamo Great idea! :+1: 

So a drag-and-drop widget can be added later to sort and hide existing apps/models right from the interface.

**Comment by merwok** [comment]
Django 4.1 lets us reorder models in the admin!  https://docs.djangoproject.com/en/4.1/ref/contrib/admin/#django.contrib.admin.AdminSite.get_app_list

**Comment by fabiocaccamo** [comment]
Oh nice, didn't know about that.
Do you think we can close this issue?
Or this app should use the `get_app_list` method for better integration?

**Comment by merwok** [comment]
I would say that defining order of models is part of making the admin adapted to specific project needs, so it’s fine to do it with code and have it be the same for all themes.  What do other people think?

**Comment by merwok** [comment]
It was actually added in 4.0 I think!

I’ve implemented it in a project with a custom AdminSite subclass implementing get_app_list using a hard-coded list of sorted app names + custom attribute `_order_` on model classes so that each app can define order for models.
It’s OK but not great.

To do it with django-admin-interface, we’d need a way to store order, for example a JSON field, and a custom AdminSite subclass that uses it.  The rainbow pony would be a fancy list widget that automatically lists all apps and models, with drag-n-drop interface, but that’s a bit of javascript work!
